### PR TITLE
signal to bug ratio post

### PR DESCRIPTION
**Pull request by shellscape**

### CODE REVIEW

File: _posts/2017-04-24-signal-to-bug-ratio.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+## Oodles of Noise
+
+Signal events are the de facto method for a client (a web page) to interact with
+a signal implementation. Typically you bind an event to `window` and use some
+JavaScript to trigger an event on `window` in the client. Sounds easy, yeah? If
+you happen to work within an environment in which an engineering team doesn't
+manage and test data being sent along with these events, debugging that `eventData`
+can be far more difficult than it should be. There's a simple solution to debugging
+`eventData` that introduces responsible patterns and easy of use, which go beyond
```
Comment (johncoghlan) [inline _posts/2017-04-24-signal-to-bug-ratio.md]: "ease of use"

File: _posts/2017-04-24-signal-to-bug-ratio.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+in any corner of our [engineering department](http://tech.gilt.com). Again, this is not to knock their
+skill level - they aren't advertised as engineers. But it's important to understand
+the distinction, and important to consult engineers for architectural needs and
+best-practice guidance.
+
+## The Console.log Conundrum
+
+At one point in time, our Signal implementation went from being an engineering-first,
+engineering-owned product, to a Marketing and Analytics owned product. With very
+little JavaScript experience they were about to work within Signal. And those
```
Comment (johncoghlan) [inline _posts/2017-04-24-signal-to-bug-ratio.md]: may want to add a comma between experience and they

File: _posts/2017-04-24-signal-to-bug-ratio.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+## Simple Engineering Save
+
+Since our implementation is not (yet!) engineering-owned, we can't run tests and
+such on the data in a pre-production environment. Some would argue that Signal has
+some pre-production ability, but even "Previewing" tags and events still runs live
+in production. So we needed a method for our Marketing and Analytics teams to be
+able to debug in a familiar way (which is unfortunately `console.log`) but in a
+_controlled_ manner that doesn't affect the world.
+
+<script src="https://gist.github.com/shellscape/fdcd98aa743b683eb4539336b06200c9.js"></script>
```
Comment (johncoghlan) [inline _posts/2017-04-24-signal-to-bug-ratio.md]: this gist isn't rendering in the preview

Comment (shellscape) [inline _posts/2017-04-24-signal-to-bug-ratio.md]: and it wont

File: _posts/2017-04-24-signal-to-bug-ratio.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+Signal events are the de facto method for a client (a web page) to interact with
+a signal implementation. Typically you bind an event to `window` and use some
+JavaScript to trigger an event on `window` in the client. Sounds easy, yeah? If
+you happen to work within an environment in which an engineering team doesn't
+manage and test data being sent along with these events, debugging that `eventData`
+can be far more difficult than it should be. There's a simple solution to debugging
+`eventData` that introduces responsible patterns and ease of use, which go beyond
+`console.log`.
+
+## Knowledge Gap
```
Comment (jkenny) [inline _posts/2017-04-24-signal-to-bug-ratio.md]: I don't think this section adds enough value to outweigh the risks in including it. Regardless of how you dress it up, it's a full section dedicated to calling out Signal's implementation consultants as non-technical. You could get the same point across by adding a small comment to the section below where you explain why anti-patterns emerge in the Signal code.

File: _posts/2017-04-24-signal-to-bug-ratio.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+skill level - they aren't advertised as engineers. But it's important to understand
+the distinction, and important to consult engineers for architectural needs and
+best-practice guidance.
+
+## The Console.log Conundrum
+
+At one point in time, our Signal implementation went from being an engineering-first,
+engineering-owned product, to a Marketing and Analytics owned product. Those
+teams did a fine job using Signal to drive the associated sites forward with very
+little JavaScript experience. However, their guidance was coming from non-engineers
```
Comment (jkenny) [inline _posts/2017-04-24-signal-to-bug-ratio.md]: How about "However, through no fault of their own (even implementation guidance from Signal themselves was often coming from non-technical people) [many anti-patterns were implemented](https://sourcemaking.com/antipatterns).

File: _posts/2017-04-24-signal-to-bug-ratio.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+as this isn't already part of Signal's client library and available for folks
+who are using the product without engineering support.
+
+Please feel free to use the script above in your implementation. Or, if you have
+a better solution, please comment!
+
+### Cheers!
+
+*Originally published at
+ [shellscape.org](http://shellscape.org/2017/04/24/signal-to-debug-ratio) on
```
Comment (gheine) [inline _posts/2017-04-24-signal-to-bug-ratio.md]: wrong link

File: _posts/2017-04-24-signal-to-bug-ratio.md (hunk 5 above)
Comment (shellscape) [inline _posts/2017-04-24-signal-to-bug-ratio.md]: mm, I don't like it as much :)

File: _posts/2017-04-24-signal-to-bug-ratio.md (hunk 4 above)
Comment (shellscape) [inline _posts/2017-04-24-signal-to-bug-ratio.md]: disagree that there are any risks in including it. calling them out is exactly what it's meant to do, intentionally prickly.

File: _posts/2017-04-24-signal-to-bug-ratio.md (hunk 6 above)
Comment (shellscape) [inline _posts/2017-04-24-signal-to-bug-ratio.md]: 👍

File: _posts/2017-04-24-signal-to-bug-ratio.md (hunk 4 above)
Comment (jkenny) [inline _posts/2017-04-24-signal-to-bug-ratio.md]: What are we trying to achieve by getting prickly with Signal on a public blog?

Comment (shellscape) [inline _posts/2017-04-24-signal-to-bug-ratio.md]: "we're" not trying to achieve anything, there's no agenda. it's the tone and timbre intended - prickly. signal is notorious for this kind of thing, and it'll speak to those who have had similar experiences with it. if you'd like to block on that no harm no foul. I syndicate related posts here to help out the tech blog with fresh content, so no worries if you'd rather not use it.

Comment (jkenny) [inline _posts/2017-04-24-signal-to-bug-ratio.md]: I'm not suggesting we should change the post to be warm and fuzzy. I'm just questioning whether the section "Knowledge Gap" is adding any value. When I read the post it felt like that section could be dropped without losing anything apart from a warning about taking technical advice from Signal's implementation consultants. Just my 2c. Feel free to leave it in if you think it's important to the post.

File: _posts/2017-04-24-signal-to-bug-ratio.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+some pre-production ability, but even "Previewing" tags and events still runs live
+in production. So we needed a method for our Marketing and Analytics teams to be
+able to debug in a familiar way (which is unfortunately `console.log`) but in a
+_controlled_ manner that doesn't affect the world.
+
+<script src="https://gist.github.com/shellscape/fdcd98aa743b683eb4539336b06200c9.js"></script>
+
+The idea is dead simple. Sprinkle `bananas.signal.debug(...)` all over, anywhere,
+go bananas! It won't output to the console until explicitly enabled. That can be
+done in two ways, but adding `?debug=true` to the URI and refreshing, or by calling
```
Comment (jkenny) [inline _posts/2017-04-24-signal-to-bug-ratio.md]: "by adding ..."

File: _posts/2017-04-24-signal-to-bug-ratio.md (hunk 4 above)
Comment (shellscape) [inline _posts/2017-04-24-signal-to-bug-ratio.md]: I really do think that folks should know they should be wary of their support. they've made so many poor decisions on our (_our_, being non-engineers at Gilt and HBC) behalf it's bananas.

File: _posts/2017-04-24-signal-to-bug-ratio.md (hunk 7 above)
Comment (shellscape) [inline _posts/2017-04-24-signal-to-bug-ratio.md]: you're a sniper! fixed.

**Pull request merged**